daemon/cmd/main.go

```diff
@@ -0,0 +1,266 @@
+		_ = level.Info(logger).Log("msg", "service started", "stackdriver.project", *flagStackDriverProject)
+	} else {
+		_ = level.Info(logger).Log("msg", "service started", "agent", *flagOCAgentAddr)
+	}
+	defer func() {
+		_ = level.Info(logger).Log("msg", "service ended")
+	}()
+
+	// Use a goroutine lifecycle manager for our services.
+	// See: https://github.com/oklog/run
+	g := &run.Group{}
+
+	// initialize and register our zPages service
+	if err := svcZPages(g, *flagZPagesAddr, *flagZPagesPathPrefix); err != nil {
+		_ = level.Error(logger).Log("msg", "unable to start zPages", "err", err)
+	}
+
+	// initialize and add our daemon message processing service
+	hnd, err := svcProcessor(g,
+		*flagMsgBufSize, *flagMsgProcCount, *flagServiceName, *flagOCAgentAddr,
+		*flagStackDriverProject, logger,
+	)
+	if err != nil {
+		_ = level.Error(logger).Log("msg", "failed to create OCAgent exporter", "err", err)
+		os.Exit(1)
+	}
+
+	// initialize and add our transport service
+	// linux/darwin: unix_socket & windows: named pipes
+	svcTransport(g, hnd, *flagTransportPath)
+
+	// initialize our signal handler for enabling graceful shutdown
+	signalHandler(g, logger)
+
+	// spawn our service goroutines and wait for shutdown
+	if err := g.Run(); err != nil {
+		_ = level.Error(logger).Log("msg", "unexpected shutdown of service", "err", err)
+	}
+}
+
+func initLogger(lvl int) (logger log.Logger) {
+	logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	logger = log.With(logger, "svc", serviceName, "ts", log.DefaultTimestampUTC, "clr", log.DefaultCaller)
+
+	switch lvl {
+	case logNone:
+		logger = log.NewNopLogger()
+	case logError:
+		logger = level.NewFilter(logger, level.AllowError())
+	case logWarn:
+		logger = level.NewFilter(logger, level.AllowWarn())
+	case logInfo:
+		logger = level.NewFilter(logger, level.AllowInfo())
+	case logDebug:
+		logger = level.NewFilter(logger, level.AllowDebug())
+	default:
+		logger = level.NewFilter(logger, level.AllowError())
+	}
+	return
+}
+
+func svcProcessor(g *run.Group, msgBufSize, msgProcCount int, svcName, ocAgentAddr, stackdriverProjectID string, logger log.Logger) (daemon.Handler, error) {
+	if msgBufSize < 100 {
+		msgBufSize = defaultMsgBufSize
+	}
+
+	if msgProcCount < 1 {
+		msgProcCount = runtime.NumCPU()
+	}
+
+	if svcName == "" {
+		svcName = serviceName
+	}
+
+	var (
+		err      error
+		exporter interface {
+			view.Exporter
+			trace.Exporter
+		}
+	)
+
+	if stackdriverProjectID != "" {
+		exporter, err = stackdriver.NewExporter(
+			stackdriver.Options{
+				ProjectID: stackdriverProjectID,
+			},
+		)
+		view.SetReportingPeriod(80 * time.Second)
+		level.Error(logger).Log("msg", "temporary exporter stackdriver enabled, agent disabled")
+	} else {
+		exporter, err = ocagent.NewExporter(
+			ocagent.WithInsecure(),
+			ocagent.WithAddress(ocAgentAddr),
+			ocagent.WithServiceName(svcName),
+		)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	// enables Daemon internal traces
```

Comment:
Not only `internal traces` but perhaps
```go
// Enables the daemon to export traces.
```

Reply:
It does only enable internal traces (and I haven't implemented any of them as so far just metrics seems to fit better for daemon observability).

The trace export is handled by the processor by calling the exporter interface directly and feeding it SpanData.